cmd/sanitize-prow-jobs/main.go

```diff
@@ -6,36 +6,120 @@ import (
+	"regexp"
+	"sort"
 	"strings"
 	"sync"
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	prowconfig "k8s.io/test-infra/prow/config"
 	"sigs.k8s.io/yaml"
 )
 
-const defaultCluster = "api.ci"
-
 type options struct {
 	prowJobConfigDir string
+	configPath       string
 
 	help bool
 }
 
+type Config struct {
+	Clusters struct {
+		Default    string `json:"default"`
+		Assignment map[string]struct {
+			// Jobs override both files and groups
+			Jobs []string `json:"jobs"`
+			// Files override files
+			Files  []string `json:"files"`
+			Groups []string `json:"groups"`
+		} `json:"assignment"`
+		Groups    map[string][]string       `json:"groups"`
+		FileREMap map[string]*regexp.Regexp `json:"-"`
+	} `json:"clusters"`
+}
+
+func (config *Config) getClusterForJob(jobName string, path string) string {
+	distribution := config.Clusters.Assignment
+	clusters := make([]string, 0, len(distribution))
+	for cluster := range distribution {
+		clusters = append(clusters, cluster)
+	}
+	sort.Strings(clusters)
+
+	//jobs override files which override groups
+	for _, cluster := range clusters {
+		for _, job := range distribution[cluster].Jobs {
+			if jobName == job {
+				return cluster
+			}
+		}
+	}
+	for _, cluster := range clusters {
+		for _, file := range distribution[cluster].Files {
+			re, ok := config.Clusters.FileREMap[file]
+			if !ok {
+				continue
+			}
+			if re.MatchString(path) {
+				return cluster
+			}
+		}
+	}
+	for _, cluster := range clusters {
+		for _, group := range distribution[cluster].Groups {
+			for _, job := range config.Clusters.Groups[group] {
+				if jobName == job {
+					return cluster
+				}
+			}
+		}
+	}
+
+	return config.Clusters.Default
+}
+
 func bindOptions(flag *flag.FlagSet) *options {
 	opt := &options{}
 
 	flag.StringVar(&opt.prowJobConfigDir, "prow-jobs-dir", "", "Path to a root of directory structure with Prow job config files (ci-operator/jobs in openshift/release)")
+	flag.StringVar(&opt.configPath, "config-path", "", "Path to the config file (core-services/sanitize-prow-jobs/_config.yaml in openshift/release)")
 	flag.BoolVar(&opt.help, "h", false, "Show help for ci-operator-prowgen")
 
 	return opt
 }
 
-func determinizeJobs(prowJobConfigDir string) error {
+func determinizeJobs(prowJobConfigDir, configPath string) error {
 	errChan := make(chan error)
 
+	config := &Config{}
+	data, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read the config file %q", configPath)
+	}
+	err = yaml.Unmarshal(data, config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to unmarshal the config %q", string(data))
+	}
+
 	var errs []error
+	fileREMap := map[string]*regexp.Regexp{}
+	for _, v := range config.Clusters.Assignment {
+		for _, file := range v.Files {
+			if _, ok := fileREMap[file]; ok {
+				continue
+			}
+			re, err := regexp.Compile(file)
+			if err != nil {
+				errs = append(errs, err)
```

Comment:
without putting the cluster name, assignmment, and file name into this error it will be impossible to debug

Reply:
Done.